packages/authgear-capacitor/src/index.ts

```diff
@@ -643,18 +667,32 @@ export class CapacitorContainer {
     return false;
   }
 
+  /**
+   * @internal
+   */
+  getScopes(): string[] {
+    const scopes = [
+      "openid",
+      "offline_access",
+      "https://authgear.com/scopes/full-access",
+    ];
+    if (this.isAppInitiatedSSOToWebEnabled) {
+      scopes.push(
+        "device_sso",
+        "https://authgear.com/scopes/app-initiated-sso-to-web"
+      );
+    }
+    return scopes;
+  }
+
   /**
    * @internal
    */
   async authorizeEndpoint(options: AuthenticateOptions): Promise<string> {
     return this.baseContainer.authorizeEndpoint({
       ...options,
       responseType: "code",
-      scope: [
-        "openid",
-        "offline_access",
-        "https://authgear.com/scopes/full-access",
-      ],
+      scope: this.getScopes(),
```

Comment:
@louischan-oursky It seems scope specified in `reauthenticate` does not really affect the scope, it only affects whether refresh_token or id_token will be issued, so I don't change it. In fact I think we don't want new offline grant to be generated in reauth so `offline_grant` should not be included.

`openSettingsAction` was also left untouched because we don't want the session created for settings can be used to perform app-initiated-sso-to-web.

Reply:
Please add your comment around those scope for further reader. That serves as a documentation on why it is written that way, and it is intended.